Kernel/Process.cpp

```diff
@@ -414,10 +414,14 @@ void signal_trampoline_dummy(void)
         "mov ebp, esp\n"
         "push eax\n"          // we have to store eax 'cause it might be the return value from a syscall
         "sub esp, 4\n"        // align the stack to 16 bytes
+        "mov eax, [ebp+20]\n" // push ucontext_t*
+        "push eax\n"
+        "mov eax, [ebp+16]\n" // push siginfo_t*
+        "push eax\n"
         "mov eax, [ebp+12]\n" // push the signal code
         "push eax\n"
         "call [ebp+8]\n" // call the signal handler
-        "add esp, 8\n"
+        "add esp, 16\n"
```

Comment:
These changes to the asm here screw up the stack alignment or something. My test program works fine, but sending things like SIGINT to a process from the shell asserts in sigaction that the restored stack ptr == the calculated one (after removing the siginfo and ucontext). Not sure how to get it  correct, since I don't understand how the 16 byte alignment requirement comes into play nor how what the original code does makes things work?

Reply:
Aha, I'm forgetting about the alignment at the top of setup_signal_stack. My assertion was wrong, not the code